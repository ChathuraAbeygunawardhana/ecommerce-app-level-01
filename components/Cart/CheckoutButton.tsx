import React from 'react';
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native';

interface CheckoutButtonProps {
  onPress: () => void;
}

const CheckoutButton = ({ onPress }: CheckoutButtonProps) => {
  return (
    <TouchableOpacity style={styles.checkoutButton} onPress={onPress}>
      <View>
        <Text style={styles.checkoutButtonText}>Checkout</Text>
      </View>
    </TouchableOpacity>
  );
};

export default CheckoutButton;

const styles = StyleSheet.create({
  checkoutButton: {
    backgroundColor: '#000',
    padding: 16,
    alignItems: 'center',
    justifyContent: 'center',
    marginBottom: 16,
    borderRadius: 8,
    width: '80%',
  },
  checkoutButtonText: {
    color: '#fff',
    fontSize: 18,
    fontWeight: 'bold',
  },
});
