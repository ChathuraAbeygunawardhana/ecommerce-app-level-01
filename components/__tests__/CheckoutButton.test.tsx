import React from 'react';
import renderer from 'react-test-renderer';
import CheckoutButton from '../Cart/CheckoutButton';

describe('CheckoutButton', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CheckoutButton onPress={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});