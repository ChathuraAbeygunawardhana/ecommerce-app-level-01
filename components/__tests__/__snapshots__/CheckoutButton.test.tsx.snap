// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutButton renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000",
      "borderRadius": 8,
      "justifyContent": "center",
      "marginBottom": 16,
      "opacity": 1,
      "padding": 16,
      "width": "80%",
    }
  }
>
  <View>
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      Checkout
    </Text>
  </View>
</View>
`;
